src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -275,32 +278,36 @@ Task HandleCompletion(BrokeredMessage message, BrokeredMessageReceiveContext con
                     return context.IncomingBrokeredMessage.SafeCompleteAsync();
                 }
             }
-            return TaskEx.Completed;
+
+            return TaskEx.CompletedTrue;
         }
 
-        Task AbandonOnCancellation(BrokeredMessage message)
+        Task<bool> AbandonOnCancellation(BrokeredMessage message)
         {
             logger.Debug("Received message is cancelled by the pipeline, abandoning it so we can process it later.");
 
             return AbandonInternal(message);
         }
 
-        Task Abandon(BrokeredMessage message, Exception exception)
+        Task<bool> Abandon(BrokeredMessage message, Exception exception)
         {
             logger.Debug("Exceptions occurred OnComplete", exception);
 
             return AbandonInternal(message);
         }
 
-        async Task AbandonInternal(BrokeredMessage message, IDictionary<string, object> propertiesToModify = null)
+        async Task<bool> AbandonInternal(BrokeredMessage message, IDictionary<string, object> propertiesToModify = null)
         {
-            if (receiveMode == ReceiveMode.ReceiveAndDelete) return;
+            if (receiveMode == ReceiveMode.ReceiveAndDelete)
+                return true;
 
+            bool wasAbandonded;
             using (var suppressScope = new TransactionScope(TransactionScopeOption.Suppress, TransactionScopeAsyncFlowOption.Enabled))
             {
                 logger.DebugFormat("Abandoning brokered message {0}", message.MessageId);
 
-                if (await message.SafeAbandonAsync(propertiesToModify).ConfigureAwait(false))
+                wasAbandonded = await message.SafeAbandonAsync(propertiesToModify).ConfigureAwait(false);
+                if (wasAbandonded)
                 {
                     logger.DebugFormat("Brokered message {0} abandoned successfully.", message.MessageId);
                 }
```

Comment:
I recall we had this discussion with @ramonsmits 
Either way, I'm not going to touch in a critical hotfix 😃